lib/editor/atto/plugins/kalturamedia/yui/build/moodle-atto_kalturamedia-button/moodle-atto_kalturamedia-button.js

```diff
@@ -15,194 +15,168 @@ YUI.add('moodle-atto_kalturamedia-button', function (Y, NAME) {
-                kafUri = this._removeProtocolFromUrl(kafUri);
-                return sourceUrl.replace(kafUri, kalturaToken);
-            },
-
-            _removeProtocolFromUrl: function(fullUrl) {
-                return fullUrl.replace(/^https?:\/\//,'');
-            },
-
-            embedItem: function(what) {
-                var dialogue = this.getDialogue({
-                    focusAfterHide: null
-                });
-
-                data = Y.one('#KalturaMediaSubmit')._getDataAttributes();
-                embedInfo = {};
-                for(param in data)
+                    width: '1112px'
+                },
+                this.get('area'));
+        return M.cfg.wwwroot + '/lib/editor/atto/plugins/kalturamedia/ltibrowse_container.php?' +
+                Y.QueryString.stringify(args);
+    },
+    
+    _getCourseId: function() {
+        var courseId;
+        var bodyClasses = document.getElementsByTagName('body')[0].className;
+        var classes = bodyClasses.split(' ');
+        for(i in classes)
+        {
+            if(classes[i].indexOf('course-') > -1)
                 {
-                    var isEmbedInfo = param.split('-');
-                    if(isEmbedInfo[0] == 'embedinfo')
-                    {
-                        embedInfo[isEmbedInfo[1]] = data[param];
-                    }
+                    var parts = classes[i].split('-');
+                    courseId = parts[1];
                 }
-
-                var token = this.get('kalturauritoken');
-                var kafUri = this.get('kafuri');
-                var sourceUrl = embedInfo.url;
-                var url = this._replaceKafUriWithToken(sourceUrl, kafUri, token);
-                var parser = document.createElement('a');
-                parser.href = sourceUrl;
-                url += parser.search;
-
-                var content = '<a href="http://'+url+'">tinymce-kalturamedia-embed||'+embedInfo.title+'||'+embedInfo.width+'||'+embedInfo.height+'</a>';
-
-                host = this.get('host');
-                host.setSelection(this._currentSelection);
-                host.insertContentAtFocusPoint(content);
-                this.markUpdated();
-                dialogue.set('bodyContent', "").hide();
+        }
+        
+        return courseId;
+    },
+
+    _replaceKafUriWithToken: function(sourceUrl, kafUri, kalturaToken) {
+        sourceUrl = this._removeProtocolFromUrl(sourceUrl);
+        kafUri = this._removeProtocolFromUrl(kafUri);
+        return sourceUrl.replace(kafUri, kalturaToken);
+    },
+
+    _removeProtocolFromUrl: function(fullUrl) {
+        return fullUrl.replace(/^https?:\/\//,'');
+    },
+
+    embedItem: function(what, data) {
+        embedInfo = {};
+        for(param in data)
+        {
+            var isEmbedInfo = param.split('-');
+            if(isEmbedInfo[0] == 'embedinfo')
+            {
+                embedInfo[isEmbedInfo[1]] = data[param];
```

Comment:
@muli I guess this index 'gambling' will repeat during this code, since i'm not familiar with the context.
isn't it possible that the expected input will be corrupted and not as expected?
if so, please add checkups before each index read. else - leave it....